pkg/registry/apis/provisioning/hello.go

```diff
@@ -42,13 +47,27 @@ func (*helloWorldSubresource) NewConnectOptions() (runtime.Object, bool, string)
 	return nil, false, ""
 }
 
-func (*helloWorldSubresource) Connect(ctx context.Context, name string, opts runtime.Object, responder rest.Responder) (http.Handler, error) {
+func (s *helloWorldSubresource) Connect(ctx context.Context, name string, opts runtime.Object, responder rest.Responder) (http.Handler, error) {
+	obj, err := s.getter.Get(ctx, name, &v1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	repo, ok := obj.(*v0alpha1.Repository)
+	if !ok {
+		return nil, fmt.Errorf("expected repository, but got %t", obj)
+	}
+
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		whom := r.URL.Query().Get("whom")
 		if whom == "" {
 			whom = "World"
 		}
 
+		fmt.Printf("GOT: %s/%s\n", repo.Name, repo.Spec.Type)
+		fmt.Printf(" local: %+v\n", repo.Spec.Local)
+		fmt.Printf(" github: %+v\n", repo.Spec.GitHub)
+		fmt.Printf(" s3: %+v\n", repo.Spec.S3)
```

Comment:
@Proximyst -- this seems to keep the nils as expected:

for http://localhost:3000/apis/provisioning.grafana.app/v0alpha1/namespaces/default/repositories/github-example/hello

I see:
```
GOT: github-example/github
 local: <nil>
 github: &{Owner:grafana Repository:git-ui-sync-demo BranchWorkflow:true GenerateDashboardPreviews:true}
 s3: <nil>
```